src/boxed.rs

```diff
@@ -162,14 +162,35 @@ impl<'a, T> Box<'a, T> {
     /// ```
     #[inline(always)]
     pub fn new_in(x: T, a: &'a Bump) -> Box<'a, T> {
-        Box(a.alloc(x))
+        Box(a.alloc(mem::ManuallyDrop::new(x)))
     }
 
     /// Constructs a new `Pin<Box<T>>`. If `T` does not implement `Unpin`, then
     /// `x` will be pinned in memory and unable to be moved.
     #[inline(always)]
     pub fn pin_in(x: T, a: &'a Bump) -> Pin<Box<'a, T>> {
-        Box(a.alloc(x)).into()
+        Box(a.alloc(mem::ManuallyDrop::new(x))).into()
+    }
+
+    /// Consumes the `Box`, returning the wrapped value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use bumpalo::{Bump, boxed::Box};
+    ///
+    /// let b = Bump::new();
+    ///
+    /// let c = Box::new_in(5, &b);
```

Comment:
I think it would be better to use a non-`Copy` type that implements `Drop` here (e.g. `String`), so that we also test that we don't run the destructor twice (Miri checks that, at least if we enable the `boxed` feature in the CI).